torch/utils/_config_module.py

```diff
@@ -184,30 +194,34 @@ def _get_dict(
         config: Dict[str, Any] = {}
         for key in self._config:
             if ignored_keys and key in ignored_keys:
-                if skip_default and self._config[key] != self._default[key]:
+                if skip_default and not self._is_default(key):
                     warnings.warn(
                         f"Skipping serialization of {key} value {self._config[key]}"
                     )
                 continue
             if ignored_prefixes:
                 if any(key.startswith(prefix) for prefix in ignored_prefixes):
                     continue
-            if skip_default and self._config[key] == self._default[key]:
+            if skip_default and self._is_default(key):
                 continue
-            config[key] = copy.deepcopy(self._config[key])
+            config[key] = copy.deepcopy(self.__getattr__(key))
         return config
 
     def save_config(self) -> bytes:
         """Convert config to a pickled blob"""
+        ignored_keys = []
+        if "_save_config_ignore" in self._config:
```

Comment:
Why didn't we just shunt this out of the config field on construction lol

Reply:
I'm probably following existing behavior too closely, in particular, this access used get (i.e. it wouldn't throw if the key was missing), versus the other accesses just directly accessing it (which will throw if the key is missing).

Similarly, this is technically mutable :).

Looking around, I don't see any examples within the pytorch codebase of this actually being updated mutable. So switching to just directly accessing the field with getattr everywhere to get a default, and I'll stack a PR to just resolve this once on install, and not put it into the configuration at all.